js/bundle.go

```diff
@@ -223,7 +223,7 @@ func (b *Bundle) Instantiate() (bi *BundleInstance, instErr error) {
 		panic("exported default is not a function")
 	}
 
-	jsOptions := rt.Get("options")
+	jsOptions := exports.Get("options")
```

Comment:
I'm not very familiar with Goja's API, but shouldn't we also conversely do `s/rt/exports/` on line 231 below?

Reply:
:facepalm: , yes ... and I looked for other instances :facepalm:

Reply:
Of course,  this will now fail :sob: ... because #722 made the test to expect that if you didn't define `options` it will be overwritten ... which might not be the worst idea ... I dunno, but we will probably need to discuss this. Also, I remember that there were some problems around this in the cloud which even more leads me to believe @na-- will need to say something on the matter ;)

Reply:
hmm I first started rebasing this without reading the comments, but looking at the test that's failing, how sure are we that this fix isn't a breaking change?

Reply:
as long as people use 
```javascript
export let options = {...}; 
```
and babel, which is ... I would go with the majority - they won't notice a thing ...
the same code with ES5.1 is 
```javascript
var options = {...};
exports.options=options; 
```

If someone decided that if you just don't make a variable `options` at all or that variable is then not set to `exports.options` and then they expect that `options` (not `exports.options`) will exist - then yes they will find out that this won't work. 
for the record I don't think this has ever happened apart from this test you wrote.

Reply:
hmm but the test was failing even when it had 
``js
var options = {...};
exports.options=options; 
```
:confused: I reverted some of the fixes, so now the failures are different, but I'll rebase again after https://github.com/loadimpact/k6/pull/1459 is merged in `new-schedulers`

Reply:
Hm, when it failed for me it was only the first [three "variants"](https://app.circleci.com/pipelines/github/loadimpact/k6/134/workflows/99a13bee-7a44-4bcd-8f4e-1ffc40441596/jobs/10019) which are :
https://github.com/loadimpact/k6/blob/35784c2498a14a06565b5d1e87ef05f721820675/js/runner_test.go#L138-L140

so.. no options or one that is defined as null or undefined ... 

I have no idea why you decided that this should work in the first place, hence the mention above for you to look at it. IMO this shouldn't have worked like that and I would be surprised if someone depended on this behavior... and if so I would argue they need to the corresponding changes :D

Reply:
> I have no idea why you decided that this should work in the first place

I think it should work, since I think a valid use case would be to run a k6 script with absolutely no exported script `options`, but to instead specify the options via environment variables or CLI flags or a JSON config or a mix of those. In that scenario, you should still be able to know with what actual options your script is being executed, from inside of the script... 

That said, if I had considered the issue carefully when I implemented it, I might have done required an exported empty `options` to exist for that feature to work... But I didn't, so now this is a breaking change that will probably break some people's scripts, especially those running k6 unsupervised :disappointed: So, I'd have to give it a :-1:, unless there is some bigger issue with the current implementation that I'm missing? What are the problems caused by just setting `options` in the global script scope, regardless of if it's exported or not? This is sort of how we do `__ENV`, `__VU` and `__ITER`, right?

Reply:
I would mostly argue that given that we read from (and only) `exports.options`, when we set options we should be setting them on `exports.options`. 

I highly doubt that this makes much of a difference for anyone writing a script, but if you believe so, I can agree to have `options` be special variable just like `__ENV`, `__VU` and `__ITER` but I would argue this will need to be also documented. And then just have it copy it from `exports.options` ... although I have to say I don't like it ...

Reply:
Yeah, I don't like it very much either... I'd be fine to make the `jsOptions := exports.Get("options")` change, but the setting of the `options` back seems like it should be either like it currently is, or to both `exports` and a global variable. 

In any case, unless I'm missing some fundamental issue here, this can keep. We've had this issue for a while and nobody has complained AFAIK, so I'm tagging this for 0.28.0

Reply:
This definitely can wait ... I just found it doing the js optimizations PR :D